src/main/java/redis/clients/jedis/Jedis.java

```diff
@@ -4023,6 +4009,23 @@ public Long xlen(final String key) {
     }
   }
 
+  @Override
+  public List<Map.Entry<String, List<StreamEntry>>> xread(final XReadParams xReadParams, final Map<String, StreamEntryID> streams) {
+    checkIsInMultiOrPipeline();
+    client.xread(xReadParams, streams);
+
+    if (!xReadParams.hasBlock()) {
+      return BuilderFactory.STREAM_READ_RESPONSE.build(client.getObjectMultiBulkReply());
+    }
+
+    client.setTimeoutInfinite();
+    try {
+      return BuilderFactory.STREAM_READ_RESPONSE.build(client.getObjectMultiBulkReply());
+    } finally {
+      client.rollbackTimeout();
```

Comment:
Does the `infiniteSoTimeout` will affect the block time? (Not sure)

Reply:
@dengliming No. It effects the socket time(out).

Reply:
https://github.com/redis/jedis/blob/f35484c9abae2350b6eed6e557830edc30d0ff50/src/test/java/redis/clients/jedis/tests/JedisTest.java#L116-L129